homeassistant/components/roomba/sensor.py

```diff
@@ -63,3 +81,107 @@ def icon(self):
     def native_value(self):
         """Return the state of the sensor."""
         return self._battery_level
+
+
+class CleaningTime(IRobotEntity, SensorEntity):
+    """Class to hold Roomba Sensor basic info."""
+
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} cleaning time total"
+
+    @property
+    def unique_id(self):
+        """Return the ID of this sensor."""
+        return f"total_cleaning_time_{self._blid}"
+
+    @property
+    def native_unit_of_measurement(self):
+        """Return the unit_of_measurement of the device."""
+        return "hours"
+
+    @property
+    def icon(self):
+        """Return the icon for the cleaning time."""
+        return "mdi:clock"
+
+    @property
+    def native_value(self):
+        """Return the state of the sensor."""
+        return self._run_stats.get("hr")
```

Comment:
Thanks for your effort on that Integration!

Wouldn't it be better to have a list of `EntityDescription`s, Instead of separate code blocks with `SensorEntity`s?
example: [sensor](https://github.com/home-assistant/core/blob/dev/homeassistant/components/zamg/sensor.py)

Another thing is you should use the builtin `UnitOfTime.HOURS` instead of "hardcoded" `hours`.

Reply:
For the current amount of sensors I don't think it is really necessary. 
There are some other components that do it the same way, with a class for each sensor, for example [minecraft_server](https://github.com/home-assistant/core/blob/dev/homeassistant/components/minecraft_server/sensor.py).

Reply:
> use the builtin UnitOfTime.HOURS

That does not work. It is not recognised by the IDE and does not work when running it either: `NameError: name 'UnitOfTime' is not defined`.

Reply:
> > use the builtin UnitOfTime.HOURS
> 
> That does not work. It is not recognised by the IDE and does not work when running it either: `NameError: name 'UnitOfTime' is not defined`.

you also need to import it at the beginning of the file
`from homeassistant.const import UnitOfTime`

Reply:
> For the current amount of sensors I don't think it is really necessary. There are some other components that do it the same way, with a class for each sensor, for example [minecraft_server](https://github.com/home-assistant/core/blob/dev/homeassistant/components/minecraft_server/sensor.py).

The use of `EntityDescription` was just a suggestion.

Reply:
Thanks, now it works perfectly.

Reply:
Thanks!